flashcards/backend/decks/viewsets.py

```diff
@@ -7,11 +10,20 @@
 from . import models, serializers
 from .models import Category
 
-
 class DeckViewSet(ModelViewSet):
     queryset = models.Deck.objects.all()
     serializer_class = serializers.DeckSerializer
     parser_classes = [MultiPartParser, JSONParser]
+    filter_backends = [DjangoFilterBackend]
+    filterset_fields = {
+        "category__name": ["exact"],
+        "difficulty_level__name": ["exact"],
+        "difficulty_level__value": ["gt", "lt"],
```

Comment:
I think it would be a good idea to create shorter names for those params. I kind of don't like the idea of showing our lookups in query params, because it shows how models are related. I would create a filterset class with shorter names for all lookups. This way instead of having `difficulty_level__name` you could have `difficulty_name`.

https://django-filter.readthedocs.io/en/stable/guide/rest_framework.html#adding-a-filterset-with-filterset-class